File: jurij_veresciaka/U1/code/php/includes/model/Clock.php
<?php

class Clock
{

    ///////////////////////////////////////////////
    //VARIABLES                                  //
    ///////////////////////////////////////////////

    private $hours;
    private $minutes;
    private $seconds;

    ///////////////////////////////////////////////
    //CONSTRUCTOR                                //
    ///////////////////////////////////////////////

    function __construct($time)
    {
        $time_array = explode(":", $time);
        $this->setHours((int)$time_array[0]);
        $this->setMinutes((int)$time_array[1]);
        $this->setSeconds((int)$time_array[2]);
    }

    ///////////////////////////////////////////////
    //HELPERS                                    //
    ///////////////////////////////////////////////

    private function expand_time_string($time)
    {
        if (strlen((string)$time) < 2) {
            $time = "0" . $time;
        }

        return $time;
    }

    ///////////////////////////////////////////////
    //CUTSOM GETTERS                             //
    ///////////////////////////////////////////////

    public function getTime()
    {
        $hours = $this->expand_time_string($this->getHours());
        $minutes = $this->expand_time_string($this->getMinutes());
        $seconds = $this->expand_time_string($this->getSeconds());

        $time = $hours . ":" . $minutes . ":" . $seconds;

        return $time;
    }

    ///////////////////////////////////////////////
    //ACCESSORS                                  //
    ///////////////////////////////////////////////

    /**
     * @param mixed $hours
     */
    public function setHours($hours)
    {
        $this->hours = $hours;
    }

    /**
     * @return mixed
     */
    public function getHours()
    {
        return $this->hours;
    }

    /**
     * @return mixed
     */
    public function getHours12Format()
    {
        return ($this->hours % 12);
    }

    /**
     * @param mixed $minutes
     */
    public function setMinutes($minutes)
    {
        $this->minutes = $minutes;
    }

    /**
     * @return mixed
     */
    public function getMinutes()
    {
        return $this->minutes;
    }

    /**
     * @param mixed $seconds
     */
    public function setSeconds($seconds)
    {
        $this->seconds = $seconds;
    }

    /**
     * @return mixed
     */
    public function getSeconds()
    {
        return $this->seconds;
    }
}

File: jurij_veresciaka/U1/code/javase/src/model/Clock.java
package model;

public class Clock {
	
    ///////////////////////////////////////////////
    //VARIABLES                                  //
    ///////////////////////////////////////////////
	
	private int hours;
	private int minutes;
	private int seconds;
	
	///////////////////////////////////////////////
    //CONSTRUCTOR                                //
    ///////////////////////////////////////////////
	
	public Clock(String time) {
		String[] timeArray = time.split(":");
		
		this.setHours(Integer.parseInt(timeArray[0]));
		this.setMinutes(Integer.parseInt(timeArray[1]));
		this.setSeconds(Integer.parseInt(timeArray[2]));
	}

    ///////////////////////////////////////////////
    //HELPERS                                    //
    ///////////////////////////////////////////////

    private String expandTimeIntToString(int timeInt)
    {
    	String timeString = Integer.toString(timeInt);
    	
        if (timeString.length() < 2) {
        	timeString = "0" + timeString;
        }

        return timeString;
    }
    
    ///////////////////////////////////////////////
    //CUTSOM GETTERS                             //
    ///////////////////////////////////////////////

    public String getTime()
    {
    	String hours = this.expandTimeIntToString(this.getHours());
    	String minutes = this.expandTimeIntToString(this.getMinutes());
    	String seconds = this.expandTimeIntToString(this.getSeconds());

        String time = hours +  ":" + minutes + ":" + seconds;

        return time;
    }
	
    ///////////////////////////////////////////////
    //ACCESSORS                                  //
    ///////////////////////////////////////////////
	
	public int getHours() {
		return this.hours;
	}
	
	public int getHours24Format() {
		return (this.hours % 12);
	}

	public void setHours(int hours) {
		this.hours = hours;
	}

	public int getMinutes() {
		return this.minutes;
	}

	public void setMinutes(int minutes) {
		this.minutes = minutes;
	}

	public int getSeconds() {
		return this.seconds;
	}

	public void setSeconds(int seconds) {
		this.seconds = seconds;
	}
}
